lib/models/patch/multi-file-patch.js

```diff
@@ -339,6 +301,70 @@ export default class MultiFilePatch {
     return false;
   }
 
+  collapseFilePatch(filePatch) {
+    this.filePatchesByMarker.delete(filePatch.getMarker());
+    for (const hunk of filePatch.getHunks()) {
+      this.hunksByMarker.delete(hunk.getMarker());
+    }
+
+    filePatch.triggerCollapseIn(this.patchBuffer);
+
+    this.filePatchesByMarker.set(filePatch.getMarker(), filePatch);
+    // This hunk collection should be empty, but let's iterate anyway just in case filePatch was already collapsed
+    for (const hunk of filePatch.getHunks()) {
+      this.hunksByMarker.set(hunk.getMarker(), hunk);
+    }
+  }
+
+  expandFilePatch(filePatch) {
+    const index = this.filePatches.indexOf(filePatch);
+
+    this.filePatchesByMarker.delete(filePatch.getMarker());
+    for (const hunk of filePatch.getHunks()) {
+      this.hunksByMarker.delete(hunk.getMarker());
+    }
+
+    const before = [];
+    let beforeIndex = index - 1;
+    while (beforeIndex >= 0) {
+      const beforeFilePatch = this.filePatches[beforeIndex];
+      before.push(...beforeFilePatch.getEndingMarkers());
+
+      if (!beforeFilePatch.getMarker().getRange().isEmpty()) {
+        break;
+      }
+      beforeIndex--;
+    }
+
+    const after = [];
+    let afterIndex = index + 1;
+    while (afterIndex < this.filePatches.length) {
+      const afterFilePatch = this.filePatches[afterIndex];
+      after.push(...afterFilePatch.getStartingMarkers());
+
+      if (!afterFilePatch.getMarker().getRange().isEmpty()) {
+        break;
+      }
+      afterIndex++;
+    }
+
+    filePatch.triggerExpandIn(this.patchBuffer, {before, after});
+
+    this.filePatchesByMarker.set(filePatch.getMarker(), filePatch);
+    for (const hunk of filePatch.getHunks()) {
+      this.hunksByMarker.set(hunk.getMarker(), hunk);
+    }
+  }
+
+  isPatchTooLargeOrCollapsed = filePatchPath => {
```

Comment:
same as above - this name is too long and the line length linters are gonna be unhappy.